src/core/src/main/java/org/apache/jmeter/gui/MainFrame.java

```diff
@@ -585,7 +585,8 @@ public void setExtendedFrameTitle(String fname) {
      * @return a panel containing the running indicator
      */
     private Component createToolBar() {
-        Box toolPanel = new Box(BoxLayout.X_AXIS);
+        JToolBar toolPanel = new JToolBar();
+        toolPanel.setFloatable(false);
```

Comment:
Oh, this indeed looks odd.

However, it looks like a single toolbar is enough, so the following should work:

```java
        JMeterToolBar toolPanel = JMeterToolBar.createToolbar(true);
        this.toolbar = toolPanel;
        GuiPackage guiInstance = GuiPackage.getInstance();
        guiInstance.setMainToolbar(toolbar);

        toolPanel.add(Box.createRigidArea(new Dimension(5, 15)));
```

Reply:
I though about doing exactly this but didn’t know whether it was on purpose to have the right side of the top panel separated from the left (from a component perspective not the visuals).

Reply:
I have no idea either, and it is not documented :-/

It looks like that was made for historical reasons only.
If we want to make the toolbar extendable, then we probably need to add explicit APIs so the plugins could contribute to the relevant sections within the toolbar.

So far I'm inclined to have a single component (which is often easier for layout) rather than a nested structure.

@pmouawad , any idea here?